src/runtime_src/core/common/debug_ip.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2021-2022 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef XRT_CORE_DEBUG_IP_H
+#define XRT_CORE_DEBUG_IP_H
+
+#include "core/common/device.h"
```

Comment:
I think you should forward declare and not include device.h and xclbin.h when it really isn't needed.
```
struct debug_ip_data;

namespace xrt_core {

class device;

namespace debug_ip {

...

}} // namespace debug_ip, xrt_core
```